File: lab8/GCD.c
/*
 * GCD.c
 * Matthew Tan
 * mxtan
 * lab 8
 * Calculates the GCD of two numbers
 */

#include<stdio.h>
#include<stdlib.h>

int main(void)
{
    int a = -1;
    int b = -1;
    int n;
    char str[100];

    printf("Enter a positive integer: ");
    while (1)
    {    
        n = scanf(" %d", &a);
        if (n != 1)
        {
            printf("Please enter a positive integer: ");
            scanf(" %s", str);
        }   
        else
        { 
            if (a < 0)
            {
                printf("Please enter a positive integer: ");
            }
            else
            {
                break;
            }
        }
    }
    
    printf("Enter another positive integer: ");
    while (1)
    {   
        n = scanf(" %d", &b);
        if (n != 1)
        {
            printf("Please enter another positive integer: ");
            scanf(" %s", str);
        }   
        else
        { 
            if (b < 0)
            {
                printf("Please enter another positive integer: ");
            }
            else
            {
                break;
            }
        }
    }

    int r;
    int gcd;
    r = a % b;
    printf("The GCD of %d ", a); 
    printf("%d", b);
    printf(" and ");
    printf(" is ");
  
    while (r > 0)
    {
        a = b;
        b = r;
        r = a % b;
    }
    gcd = b;
    printf(" %d", gcd);
    printf(" \n");
    return 0;
} 

File: pa3/GCD.java
// GCD.java
// Matthew Tan
// mxtan
// pa3
// This calculates the GCD of two numbers

import java.util.*;

public class GCD
{
  public static void main(String[] args)
  {
    Scanner scan = new Scanner(System.in);
    int a;
    int b;
    
    System.out.print("Enter a positive integer: ");
    while (true)
    {
      if (scan.hasNextInt())
      {
        a = scan.nextInt();
        if (a > 0)
        {
          break;
        }
      }
      else
      {
        scan.next();
      }
      System.out.print("Please enter a positive integer: ");
    }
    
    System.out.print("Enter another positive integer: ");
    while (true)
    {
      if (scan.hasNextInt())
      {
        b = scan.nextInt();
        if (b > 0)
        {
          break;
        }
      }
      else
      {
        scan.next();
      }
      System.out.print("Please enter a  positive integer: ");
    }

    int r;
    int gcd;
    r = a % b;
    System.out.print("The GCD of " + a +  " and " + b + " is ");
		
    while(r > 0) 
    { 
      a = b;
      b = r;
      r = a % b;
    }
    gcd = b;		
    System.out.println(gcd);
  }
}


